files/private-chef-ctl-commands/key_control.rb

```diff
@@ -0,0 +1,325 @@
+#
+# Copyright 2015 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "openssl"
+require "optparse"
+require "ostruct"
+require "helpers/key_ctl_helper"
+
+add_command_under_category "add-client-key", "key-rotation", "Create a new client key", 2 do
+  @helper = KeyCtlHelper.new
+  @options = OpenStruct.new
+  @options.expiration_date = "infinity"
+  @key = nil
+  @usage = "Usage: chef-server-ctl add-client-key -o, --orgname NAME -c, --clientname NAME -p, --public-key-path PATH [-e, --expiration-date DATE] [-k, --key-name NAME]"
```

Comment:
Hrm did more research on how optparse works...that would totally be doable. Lemme look into how much work it would be now that I have written a zillon tests :S.